lsp-ui-doc.el

```diff
@@ -773,14 +773,40 @@ It is supposed to be called from `lsp-ui--toggle'"
 
 (defun lsp-ui-doc--glance-hide-frame ()
   "Hook to hide hover information popup for lsp-ui-doc-glance."
-  (lsp-ui-doc-hide)
-  (remove-hook 'pre-command-hook 'lsp-ui-doc--glance-hide-frame))
+  (when (or (overlayp lsp-ui-doc--inline-ov)
+            (let ((frame (lsp-ui-doc--get-frame)))
+              (and frame (frame-visible-p frame)
+                   (not (frame-focus-state frame)))))
```

Comment:
Maybe make this a lambda ? This is a little hard to read.

Reply:
Can you give me an example of how should I change this?

Reply:
I would put the and condition in a lambda similiar to
```
(let ((func (lambda (frame)
           (and frame (frame-visible-p frame)
                (not (frame-focus-state frame)))))
          (frame (lsp-ui-doc--get-frame)))
(funcall func frame)
(when ...)
```
and put the lambda function where now the let begins. 
Sorry I just woke up, but I think you get what I mean =)

Reply:
there is `cl-letf` which is better than let + funcall.

Reply:
Updated!

Reply:
Updated!

Reply:
I don't really see the difference in this case. I thought `cl-letf` is used to temporarily override a function defintion.

Reply:
Its my mistake - I was over the phone - the correct macro is cl-flet.

Reply:
Both of them work, however using cl-letf allow additional normal let binding in the same command, so it would prefer it over `cl-flet`. Also since `cl-flet` doesn't do recursive binding, `cl-labels` may be a better choice for that case.